### DISCUSSION TITLE
Add Links in SyntenyTrack MCScanAnchorsAdapter feature details

### DISCUSSION THREAD

**Discussion by srobb1** [GMOD/jbrowse-components#4591] [General] [answered]
Hello!

I would like to make links in the feature details of a mcscan anchors track to reload the two linear browsers to the core feature coords and the mate feature coords.

Essentially, I'd like to 
1. click on the CCA3H2 MCSCAN-anchor-feature ("CCA3H2-CCA3H1-orthologs" feature) to get the details, as is displayed. 
2. Then click on the Core details Position ({CCA3H2}Scaffold_5__2_contigs__length_229678556:208,102,934..208,174,803 (-))  and/or "Name" (CCA3h2_Scaffold_5__2_contigs__length_229678556_001325.1) to reload the top linear view to zoom into the gene feature
3. Then click the Mate details Position ({CCA3H1}Scaffold_5__6_contigs__length_225558114:21,320,316..21,393,044 (+)) and/or Name (CCA3h1_Scaffold_5__6_contigs__length_225558114_000208.1) to reload the bottom linear view to zoom into the mate gene feature.
4. or one button that says zoom into the core and mate coords in their corresponding browsers.

I see how I can manage this in a [simple track](https://jbrowse.org/jb2/docs/config_guides/customizing_feature_details/) , but am a bit confused about how to handle links to two different browser windows. 

Thank you for your help,
Sofia

![Screenshot 2024-09-25 at 1 53 57 PM](https://github.com/user-attachments/assets/163f768c-2acf-40e5-bc89-fec15abbe1b4)

**Comment by srobb1** [comment]
Maybe issue #4482 and this [tutorial](https://jbrowse.org/jb2/docs/config_guides/customizing_feature_details/) answers my question. 

I am new to custom config in JB2. How would I get the feature's Mate name? 

Can you use url_params with the ID of a feature in loc=?

I am guessing I can use something like: 
- mate.assemblyName to get the mate assembly name.
- assemblyName to get the core assembly name
- mate.name to get the mate gene feature id
- name to get the core gene feature id

Thanks,
Sofia

**Reply by cmdcolin** [reply]
this is a little roundabout way to answer this question but...i think this could make a good 'core feature' and in that case, might be something that could just be implemented by default on all synteny feature detail panels. unless there is something particular to your use case, i imagine just adding this to the "jbrowse core" codebase would be great.

aside: I have been trying to create some sort of similar features for launching/controlling a "breakpoint split view" (for investigating structural variants) directly. An example of this recent effort is pictured in the screenshot pictured here https://github.com/GMOD/jbrowse-components/pull/4561#issue-2520636403  (see the "top/bottom" or "single row", those are two different modes to visualize the SV). I imagine working on some similar efforts to help launch/control the "synteny view" from the feature details panel would be valuable as well

as far as actually answering your questions

yes. in the feature details, for "synteny features" at least, there is 

- feature.assemblyName
- feature.start
- feature.end
- feature.refName
- feature.mate.assemblyName
- feature.mate.refName
- feature.mate.start
- feature.mate.end 
I would probably not use the feature.name or feature.mate.name for "lookup" in the ?loc= system as it is more straightforward for jbrowse generally to just input the coordinates as a locstring (refname:start-end) because then it doesn't have to lookup names in the text search index which can cause extra slowness or complication at the very least :)

**Reply by cmdcolin** [reply]
(note: in how i was conceiving of it, it wouldn't necessarily pop open a new browser tab)

**Comment by srobb1** [comment]
Hi Colin,
It would be a great core feature! I would love it! It would really help navigating around syntenic regions, it is a bit of a bear right now (but I am happy to have the syntenic capabilities!!) .  Thanks for the info about retrieving data and the input on the efficiency of the loc look up.

Sofia

**Reply by cmdcolin** [reply]
I tried to make a pass at implementing this here https://github.com/GMOD/jbrowse-components/pull/4624

it adds a link at the bottom of the feature details to "Center view on feature" (if you click on a feature in an existing synteny view) and also has a link to "Launch a new synteny view" (if you click on a synteny feature outside of a synteny view, e.g. a plain old linear genome view can load synteny tracks)

**Reply by cmdcolin** [reply, marked as answer]
released this feature in v2.16.1. if you get a chance to try it, let me know if it helps :)

**Comment by srobb1** [comment]
I have been playing with this. I like it! And I have played with the multi genome syn view. That is nice too!

Thanks!
